examples/dispatch.js

```diff
@@ -27,9 +27,10 @@ export default async function() {
     const contacts = page.locator('a[href="/contacts.php"]');
     await contacts.dispatchEvent("click");
 
-    const h3 = page.locator("h3");
-    const ok = await h3.textContent() == "Contact us";
-    check(ok, { "header": ok });
+    await check(page.locator('h3'), {
+      'header': async lo => lo.textContent()
+        .then(text => text == 'Contact us')
+    });
```

Comment:
I can agree with doing mutations inside the check-functions.

But I wouldn't consider using `then` an expert-level feature. 

I actually prefer it in some situations, such as when you're just doing a series of transformations on the value. Sort of like you would with an array:

```ts
const names = myArray
  .filter(item => item.isActive)
  .map(item => item.name)

const userName = await fetch("https://fake.api/users/1")
  .then(response => response.json())
  .then(response => response.userName)
```

But maybe that's because I'm an old goat who remember when `Promise`s promised to save us from callback hell. 😅

Reply:
I got rid of `then`s to make it simpler. WDYT now @ankur22?